### PR TITLE
intel_adsp: Add board definitions for adsp simulator

### PR DESCRIPTION
**Pull request by ceolin**
Add board definition for Intel ADSP (ACE family) simulators.

### CODE REVIEW

**Review by kv2019i — DISMISSED** [review]

**Comment by nashif** [comment]
you need to fix the ci issues...

**Comment by ceolin** [comment]
> you need to fix the ci issues...

working on it

**Review by nashif — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

File: boards/intel/adsp/intel_adsp_ace30_ptl_sim.dts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
```
Comment (dkalowsk) [inline boards/intel/adsp/intel_adsp_ace30_ptl_sim.dts:2]: I don't see this as a moved file, usually copyrights are for when the file is introduced.

File: soc/intel/intel_adsp/ace/multiprocessing.c [hunk 2]
```diff
@@ -83,6 +83,13 @@ __imr void soc_num_cpus_init(void)
 
 void soc_mp_init(void)
 {
+#if defined(CONFIG_INTEL_ADSP_SIM_NO_SECONDARY_CORE_FLOW)
+	/* BADDR stores the Xtensa LX7 AltResetVec input */
+	for (int i = 0; i < soc_num_cpus; i++) {
+		DSPCS.bootctl[i].baddr = (uint32_t)z_soc_mp_asm_entry;
+	}
+#endif
```
Comment (dkalowsk) [inline soc/intel/intel_adsp/ace/multiprocessing.c:91]: missing closing comment here

File: boards/intel/adsp/Kconfig.intel_adsp [hunk 3]
```diff
@@ -5,5 +5,8 @@ config BOARD_INTEL_ADSP
 	select SOC_INTEL_CAVS_V25 if BOARD_INTEL_ADSP_CAVS25
 	select SOC_INTEL_CAVS_V25 if BOARD_INTEL_ADSP_CAVS25_TGPH
 	select SOC_INTEL_ACE15_MTPM if BOARD_INTEL_ADSP_ACE15_MTPM
+	select SOC_INTEL_ACE15_MTPM if BOARD_INTEL_ADSP_ACE15_MTPM_SIM
```
Comment (dkalowsk) [inline boards/intel/adsp/Kconfig.intel_adsp:8]: would this be better as:
```
	select SOC_INTEL_ACE15_MTPM if BOARD_INTEL_ADSP_ACE15_MTPM || 
                                                                    BOARD_INTEL_ADSP_ACE15_MTPM_SIM
```
Don't mind the spacing that was done just to make the examples here easier.  Thought being it'll introduce less risk if the SOC needs to change at some point.

**Review by dkalowsk — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**